src/openvic-simulation/map/Map.cpp

```diff
@@ -609,15 +573,13 @@ bool Map::load_map_images(fs::path const& province_path, fs::path const& terrain
 /* REQUIREMENTS:
  * MAP-19, MAP-84
  */
-bool Map::_generate_province_adjacencies() {
+bool Map::_generate_standard_province_adjacencies() {
 	bool changed = false;
 
-	auto generate_adjacency = [&](Province* current, size_t x, size_t y) -> bool {
+	const auto generate_adjacency = [this, &changed](Province* current, size_t x, size_t y) -> bool {
 		Province* neighbour = get_province_by_index(province_shape_image[x + y * width].index);
-		if (neighbour != nullptr && current != neighbour) {
-			const Province::distance_t distance = current->calculate_distance_to(neighbour);
-			const Province::flags_t flags = 0;
-			return current->add_adjacency(neighbour, distance, flags) | neighbour->add_adjacency(current, distance, flags);
+		if (neighbour != nullptr) {
+			return Province::add_standard_adjacency(*current, *neighbour);
```

Comment:
Why not let the province instance handle it?
`current.add_standard_adjacency(neighbour)`

Reply:
The previous system required calling the function twice: `current.add_standard_adjacency(neighbour)` and `neighbour.add_standard_adjacency(current)`. I changed this to a single function that adds the adjacency in both directions, avoiding the need to do various checks and calculations twice, and made that function static/take both provinces as arguments to make it clear that it wasn't just adding the adjacency to the left province argument. It also makes the function easier to read since you have stuff like `from.is_water() || to.is_water()` instead of `is_water() || to.is_water()`.